msal/application.py

```diff
@@ -757,6 +780,11 @@ def _build_client(self, client_credential, authority, skip_regional_client=False
             # Use client_credential.get("...") rather than "..." in client_credential
             # so that we can ignore an empty string came from an empty ENV VAR.
             if client_credential.get("client_assertion"):
+                if isinstance(client_credential["client_assertion"], str):
+                    warnings.warn(
+                        "Please use a callback instead of a string. See also "
```

Comment:
```suggestion
                        "Use a callback instead of a string. MSAL will invoke the callback everytime it needs to get a token. The callback is responsible for returning non-expired assertion and for caching assertions. See also "
```

Reply:
Will accept this suggestion in my next commit.

We might as well also append "... with the help of a helper `AutoRefresher`. See also ...".

I'm providing more explanation about the AutoRefresher in the [last comment below](https://github.com/AzureAD/microsoft-authentication-library-for-python/pull/747#discussion_r1768459705).